language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/BallerinaLanguageServer.java

```diff
@@ -112,8 +114,15 @@ public CompletableFuture<InitializeResult> initialize(InitializeParams params) {
         final InitializeResult res = new InitializeResult(new ServerCapabilities());
         res.getCapabilities().setTextDocumentSync(TextDocumentSyncKind.Full);
 
+        Map initializationOptions = null;
```

Comment:
We can move the following lines before this line:
```java
        TextDocumentClientCapabilities textDocClientCapabilities = params.getCapabilities().getTextDocument();
        WorkspaceClientCapabilities workspaceClientCapabilities = params.getCapabilities().getWorkspace();
        LSClientCapabilities capabilities = new LSClientCapabilitiesImpl(textDocClientCapabilities,
                workspaceClientCapabilities,
                experimentalClientCapabilities,
                initializationOptions);
        this.serverContext.put(LSClientCapabilities.class, capabilities);
```

And then check 
```
    if (capabilities.getInitializationOptions().isLisghtWeightMode()) {
            return CompletableFuture.supplyAsync(() -> res);
    }
```
And we can remove the `isLightWeightMode()` method as well